test/pipelines/json_encoder_test.exs

```diff
@@ -6,7 +6,7 @@ defmodule Pipelines.JSONEncoderTest do
   test "Converts a given map to a json string" do
     pipelines = [Crawly.Pipelines.JSONEncoder]
     item = @valid
-    state = %{}
+    state = %{spider_name: Test, crawl_id: "test"}
```

Comment:
same as the middlewares, testing for crawl_id should be in the data storage worker test file